src/Mbstring/Mbstring.php

```diff
@@ -301,7 +301,8 @@ public static function mb_convert_case($s, $mode, $encoding = null)
                 $map = $upper;
             } else {
                 if (self::MB_CASE_FOLD === $mode) {
-                    $s = str_replace(self::CASE_FOLD[0], self::CASE_FOLD[1], $s);
+                    $caseFolding = self::getData('caseFolding');
+                    $s = str_replace(\array_keys($caseFolding), \array_values($caseFolding), $s);
```

Comment:
This is sub-optimal: the map shouldn't require any processing to be used.
Please adjust the map so that this line can be written as:
`$s = str_replace($caseFolding[0], $caseFolding[1], $s);`

Reply:
alternatively, we could try using strtr btw
if you can bench which variant would be fastest, that'd be great!

Reply:
Thank you for your suggestions! Here the benchmark code:

```php
$clock = new MonotonicClock();

$caseFolding = require 'map.php';
$keys = \array_keys($caseFolding);
$values = \array_values($caseFolding);

$now = $clock->now();
$s = str_replace($keys, $values, 'HI ßΐᾬﬗ');
$time = $clock->now()->diff($now);

$output->writeln($s.' str_replace µs :');
dump($time->f);

$now = $clock->now();
$s = strtr('HI ßΐᾬﬗ', $caseFolding);
$time = $clock->now()->diff($now);

$output->writeln($s.' strtr µs :');
dump($time->f);
```

Here are the results of a few executions:

| Attempt  | `str_replace` (µs) | `strtr` (µs) |
| --------- | ------------- | ------ |
| 1  | 5.8E-5  | 4.0E-6 |
| 2  | 5.5E-5  | 4.0E-6 |
| 3  | 1.1E-5  | 4.0E-6 |
| 4  | 5.4E-5  | 3.0E-6 |
| 5  | 1.1E-5  | 4.0E-6 |

Because of these results, I suggest to use `strtr`. I updated the PR accordingly! 👍